src/Traits/BelongsToThrough.php

```diff
@@ -13,14 +13,15 @@ trait BelongsToThrough
      *
      * @template TRelatedModel of \Illuminate\Database\Eloquent\Model
      * @template TIntermediateModel of \Illuminate\Database\Eloquent\Model
+     * @template TIntermediateModels of list<TIntermediateModel>
      *
      * @param class-string<TRelatedModel> $related
      * @param class-string<TIntermediateModel>[]|array{0: class-string<TIntermediateModel>, 1: string}[]|class-string<TIntermediateModel> $through
```

Comment:
@calebdw the main issue remaining is resolving the relation between `TIntermediateModel` and `TIntermediateModels`

Reply:
I understand, but how do we do that? We need to link the `$through` param to `TIntermediateModels`, otherwise phpstan can't track which models go in to form `TIntermediateModels`

Reply:
Oh that's easy (or maybe not, but try):

```suggestion
     * @param TIntermediateModels $through
```

Reply:
The intermediate models might be too complicated to solve through the use of phpdocs alone---we might need to create a PHPStan extension for this

Reply:
`@param TIntermediateModels $through` doesn't work, results in a bunch of errors because inference is lost. This is a fun challenge, but also a bit above my generics/PHPStan skill level atm 😅

Reply:
Hmmm, this will need an extension then as there's no detect multiple different models in the same parameter (the best that you could hope for is a union)

Reply:
What do we do here? Can we go back to the previous state by only reverting a few changes? Or is this a critical issue that needs to be fixed and/or are too many other changes connected to this?

Writing an extension sounds a lot of work and I don't want you to spend too much time on this.

Reply:
The easiest thing to do would be to just stick to the base Model---the intermediate models aren't that important, what really matters is the related model.

The extension wouldn't be that much work, maybe two classes that could be adapted from Larastan. However, it might be overkill as again I don't think they're that important

Reply:
@staudenmeir and @calebdw I applied the suggested change